sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,358 @@
+     * "application/json" type. Note that "application/json" is still a possible value for this field.
+     * @return the content type the data is in, or null if the data is nonexistent or in "application/json" format.
+     */
+    public String getDataContentType() {
+        return this.dataContentType;
+    }
+
+    /**
+     * Get the schema that the data adheres to.
+     * @return a URI of the data schema, or null if it is not set.
+     */
+    public String getDataSchema() {
+        return this.dataSchema;
+    }
+
+    /**
+     * Set the schema that the data adheres to.
+     * @param dataSchema a URI identifying the schema of the data.
+     *
+     * @return the cloud event itself.
+     */
+    public CloudEvent setDataSchema(String dataSchema) {
+        this.dataSchema = dataSchema;
+        return this;
+    }
+
+    /**
+     * Get the subject associated with this event.
+     * @return the subject, or null if the subject was not set.
+     */
+    public String getSubject() {
+        return this.subject;
+    }
+
+    /**
+     * Set the subject of the event.
+     * @param subject the subject to set.
+     *
+     * @return the cloud event itself.
+     */
+    public CloudEvent setSubject(String subject) {
+        this.subject = subject;
+        return this;
+    }
+
+    /**
+     * Get the spec version of this CloudEvent
+     * @return the spec version, or null if the spec version was not set.
+     */
+    public String getSpecVersion() {
+        return specVersion;
+    }
+
+    /**
+     * Get a map of the additional user-defined attributes associated with this event.
+     * @return the extension attributes as an unmodifiable map.
+     */
+    @JsonAnyGetter
+    public Map<String, Object> getExtensionAttributes() {
+        return this.extensionAttributes;
+    }
+
+    /**
+     * Add/Overwrite a single extension attribute to the cloud event. The property name will be transformed
+     * to lowercase and must not share a name with any reserved cloud event properties.
+     * @param name  the name of the attribute.
+     * @param value the value to associate with the name.
+     *
+     * @return the cloud event itself.
+     */
+    @JsonAnySetter
+    public CloudEvent addExtensionAttribute(String name, Object value) {
+        if (!validateAttributeName(name)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "'name' must have at most 20 alphanumeric characters and not be one of the CloudEvent attribute names"));
+        }
+        if (this.extensionAttributes == null) {
+            this.extensionAttributes = new HashMap<>();
+        }
+        this.extensionAttributes.put(name.toLowerCase(Locale.ENGLISH), value);
+        return this;
+    }
+
+    private static boolean validateAttributeName(String name) {
+        if (name.length() > 20) {
```

Comment:
The 20 character limit is mentioned as a SHOULD NOT rather than MUST NOT which means it is not recommended but still allowed.

Reply:
CNCF CloudEvent Java SDK has this validation. Anyway, to make it consistent across all languages of Azure EG SDKs, I've removed it from Java.